packages/typespec-go/test/azure-http-specs/azure/resource-manager/methodsubscriptionidgroup/utils_test.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package methodsubscriptionidgroup_test
+
+import (
+	"context"
+	"fmt"
+	"methodsubscriptionidgroup"
+	"net/http"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	azfake "github.com/Azure/azure-sdk-for-go/sdk/azcore/fake"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+var (
+	ctx           context.Context
+	clientFactory *methodsubscriptionidgroup.ClientFactory
+
+	subscriptionIdExpected = "00000000-0000-0000-0000-000000000000"
+)
+
+func TestMain(m *testing.M) {
+	ctx = context.Background()
+	clientFactory, _ = methodsubscriptionidgroup.NewClientFactory(subscriptionIdExpected, &azfake.TokenCredential{}, &arm.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			InsecureAllowCredentialWithHTTP: true,
+			PerCallPolicies: []policy.Policy{
```

Comment:
This can be simplified by setting the endpoint via cloud config.
```go
		ClientOptions: azcore.ClientOptions{
			Cloud: cloud.Configuration{
				Services: map[cloud.ServiceName]cloud.ServiceConfiguration{
					cloud.ResourceManager: {
						Audience: "fake_audience",
						Endpoint: "http://localhost:3000",
					},
				},
			},
			InsecureAllowCredentialWithHTTP: true,
		},
```